lib/elastic_apm/span/context/destination.rb

```diff
@@ -5,21 +5,37 @@ class Span
     class Context
       # @api private
       class Destination
-        def initialize(name: nil, resource: nil, type: nil)
+        def initialize(
+          name: nil,
+          resource: nil,
+          type: nil,
+          address: nil,
+          port: nil
+        )
           @name = name
           @resource = resource
           @type = type
+          @address = address
+          @port = port
         end
 
-        attr_reader :name, :resource, :type
+        attr_reader(
+          :name,
+          :resource,
+          :type,
+          :address,
+          :port
+        )
 
-        def self.from_uri(uri_or_str, type: 'external')
+        def self.from_uri(uri_or_str, type: 'external', port: nil)
```

Comment:
I added this argument, in case we are able to pass in a default port at some point.